src/jenkins_job_insight/storage.py

```
@@ -2692,3 +2709,173 @@ async def get_user_tokens(username: str) -> dict[str, str]:
+        conditions.append("tier = ?")
+        params.append(tier)
+    if version:
+        conditions.append("version = ?")
+        params.append(version)
+
+    where = " AND ".join(conditions) if conditions else "1=1"
+
+    async with aiosqlite.connect(DB_PATH) as db:
+        db.row_factory = aiosqlite.Row
+        cursor = await db.execute(
+            f"SELECT job_name, team, tier, version, labels FROM job_metadata WHERE {where} ORDER BY job_name",
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)